File: exception/src/main/java/yun/blog/exception/member/MemberService.java
package yun.blog.exception.member;

import java.io.IOException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;


@Service
@RequiredArgsConstructor
@Transactional
public class MemberService {

  private final MemberRepository memberRepository;

  public Member createUncheckedException() {
    final Member member = memberRepository.save(new Member("yun"));
    if (true) {
      throw new RuntimeException();
    }
    return member;
  }

  public Member createCheckedException() throws IOException {
    final Member member = memberRepository.save(new Member("wan"));
    if (true) {
      throw new IOException();
    }
    return member;
  }

  public Member findById(long id) {
    return memberRepository.findById(id).get();
  }

}

File: kotlin-junit5/src/main/kotlin/com/example/kotlinjunit5/member/MemberService.kt
package com.example.kotlinjunit5.member

import org.springframework.stereotype.Service

@Service
class MemberService(
    private val memberRepository: MemberRepository
) {

    fun updateName(name: String, id: Long): Member {
        val member = memberRepository.findById(id).orElseThrow { IllegalArgumentException("$id is not existed") }
        memberRepository.findByName(name)
        member.updateName(name)
        return member
    }
}